src/v/kafka/server/handlers/alter_configs.cc

```diff
@@ -69,50 +71,44 @@ create_topic_properties_update(
      * configuration in topic table, the only difference is the replication
      * factor, if not set in the request explicitly it will not be overriden.
      */
-    update.properties.compaction_strategy.op
-      = cluster::incremental_update_operation::set;
-    update.properties.compression.op
-      = cluster::incremental_update_operation::set;
-    update.properties.segment_size.op
-      = cluster::incremental_update_operation::set;
-    update.properties.timestamp_type.op
-      = cluster::incremental_update_operation::set;
-    update.properties.retention_bytes.op
-      = cluster::incremental_update_operation::set;
-    update.properties.retention_duration.op
-      = cluster::incremental_update_operation::set;
-    update.properties.shadow_indexing.op
-      = cluster::incremental_update_operation::set;
-    update.custom_properties.replication_factor.op
-      = cluster::incremental_update_operation::none;
-    update.custom_properties.data_policy.op
-      = cluster::incremental_update_operation::none;
+    update.properties.compaction_strategy.op = op_t::remove;
```

Comment:
I wonder if it's worth looping over `serde_fields`  and then fixing up the odd ones out?
```c++
    using op_t = cluster::incremental_update_operation;
    constexpr auto apply_op = [](op_t op) {
        return [op](auto&&... prop) { ((prop.op = op), ...); };
    };
    std::apply(apply_op(op_t::remove), update.properties.serde_fields());
    std::apply(apply_op(op_t::none), update.custom_properties.serde_fields());

    update.properties.cleanup_policy_bitflags.op = op_t::set;
    update.properties.cleanup_policy_bitflags.value
      = ctx.metadata_cache().get_default_cleanup_policy_bitflags();
```

Reply:
Maybe it's an abuse of `serde_fields`, and not naming them makes it harder to discover what code needs to be written when searching for in order to implement another property.

But it might still be worth using `std::tie` and naming them like:
```c++
    default_remove = std::tie(
      update.properties.foo,
     ...
    )
```
and then it might also be wroth adding a static assert that the sum of the named fields is equal to the size of the sum of the `serde_fields` of the `poperties` and `custom_properties`.

Reply:
🔥 yeah I've adopted this now

Reply:
I think it's worth making it clear if a new field is added, that maybe something needs to be consisdered here, perhaps something like:
```c++
    static_assert(
      std::tuple_size_v<decltype(update.properties.serde_fields())> == 26,
      "If you added a property, please decide on it's default alter config "
      "policy");
    static_assert(
      std::tuple_size_v<decltype(update.custom_properties.serde_fields())> == 2,
      "If you added a property, please decide on it's default alter config "
      "policy");
```

Reply:
That's along the lines of what I am doing in the follow-up PR: https://github.com/redpanda-data/redpanda/pull/21325. Let me add these there.

Reply:
I don't see the check for `update.custom_properties`.

Reply:
I've now added these checks in this PR.